test/stores/VDocument.test.ts

```diff
@@ -1088,5 +1088,223 @@ describe('stores', () => {
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>ab[]ef</p>',
+                });
+                // Backward
+                testEditor({
+                    contentBefore: '<p>ab]cd[ef</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>ab[]ef</p>',
+                });
+            });
+            it('should delete across two paragraphs', () => {
+                // deleteForward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p>ab[cd</p><p>ef]gh</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>ab[]gh</p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p>ab]cd</p><p>ef[gh</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>ab[]gh</p>',
+                });
+                // deleteBackward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p>ab[cd</p><p>ef]gh</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>ab[]gh</p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p>ab]cd</p><p>ef[gh</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>ab[]gh</p>',
+                });
+            });
+            it('should delete all the text in a paragraph', () => {
+                // deleteForward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p>[abc]</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>[]<br></p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p>]abc[</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>[]<br></p>',
+                });
+                // deleteBackward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p>[abc]</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>[]<br></p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p>]abc[</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>[]<br></p>',
+                });
+            });
+            it('should delete a complex selection accross format nodes and multiple paragraphs', () => {
+                // deleteForward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p><b>ab[cd</b></p><p><b>ef<br/>gh</b>ij<i>kl]</i>mn</p>',
```

Comment:
Could you select until `k` (included) such that there is still a `<i>` in the result ? I think this would make the test (and its derivatives) even more interesting.

Reply:
Why not do both?

Reply:
Did this:
```typescript
it('should delete a complex selection accross format nodes and multiple paragraphs', () => {
    // deleteForward
    // Forward selection
    testEditor({
        contentBefore: '<p><b>ab[cd</b></p><p><b>ef<br/>gh</b>ij<i>kl]</i>mn</p>',
        stepFunction: deleteForward,
        contentAfter: '<p><b>ab[]</b>mn</p>',
    });
    testEditor({
        contentBefore: '<p><b>ab[cd</b></p><p><b>ef<br/>gh</b>ij<i>k]l</i>mn</p>',
        stepFunction: deleteForward,
        contentAfter: '<p><b>ab[]</b><i>l</i>mn</p>',
    });
    // Backward selection
    testEditor({
        contentBefore: '<p><b>ab]cd</b></p><p><b>ef<br/>gh</b>ij<i>kl[</i>mn</p>',
        stepFunction: deleteForward,
        contentAfter: '<p><b>ab[]</b>mn</p>',
    });
    testEditor({
        contentBefore: '<p><b>ab]cd</b></p><p><b>ef<br/>gh</b>ij<i>k[l</i>mn</p>',
        stepFunction: deleteForward,
        contentAfter: '<p><b>ab[]</b><i>l</i>mn</p>',
    });
    // deleteBackward
    // Forward selection
    testEditor({
        contentBefore: '<p><b>ab[cd</b></p><p><b>ef<br/>gh</b>ij<i>kl]</i>mn</p>',
        stepFunction: deleteBackward,
        contentAfter: '<p><b>ab[]</b>mn</p>',
    });
    testEditor({
        contentBefore: '<p><b>ab[cd</b></p><p><b>ef<br/>gh</b>ij<i>k]l</i>mn</p>',
        stepFunction: deleteBackward,
        contentAfter: '<p><b>ab[]</b><i>l</i>mn</p>',
    });
    // Backward selection
    testEditor({
        contentBefore: '<p><b>ab]cd</b></p><p><b>ef<br/>gh</b>ij<i>kl[</i>mn</p>',
        stepFunction: deleteBackward,
        contentAfter: '<p><b>ab[]</b>mn</p>',
    });
    testEditor({
        contentBefore: '<p><b>ab]cd</b></p><p><b>ef<br/>gh</b>ij<i>k[l</i>mn</p>',
        stepFunction: deleteBackward,
        contentAfter: '<p><b>ab[]</b><i>l</i>mn</p>',
    });
});
```
OK?